torch_geometric_temporal/nn/convolutional/mtgnn.py

```diff
@@ -0,0 +1,445 @@
+        self._time_conv = nn.ModuleList()
+        self._kernel_set = [2, 3, 6, 7]
+        c_out = int(c_out/len(self._kernel_set))
+        for kern in self._kernel_set:
+            self._time_conv.append(nn.Conv2d(c_in, c_out, (1, kern),
+                                        dilation=(1, dilation_factor)))
+
+    def forward(self, X_in: torch.FloatTensor) -> torch.FloatTensor:
+        """
+        Making a forward pass of dilated inception.
+
+        Arg types:
+            * **X_in** (Pytorch Float Tensor) - Input feature Tensor, with shape (batch_size, c_in, num_nodes, seq_len).
+
+        Return types:
+            * **X** (PyTorch Float Tensor) - Hidden representation for all nodes, 
+            with shape (batch_size, c_out, num_nodes, seq_len-6).
+        """
+        X = []
+        for i in range(len(self._kernel_set)):
+            X.append(self._time_conv[i](X_in))
+        for i in range(len(self._kernel_set)):
+            X[i] = X[i][..., -X[-1].size(3):]
+        X = torch.cat(X, dim=1)
+        return X
+
+
+class GraphConstructor(nn.Module):
+    r"""An implementation of the graph learning layer to construct an adjacency matrix.
+    For details see this paper: `"Connecting the Dots: Multivariate Time Series Forecasting with Graph Neural Networks." 
+    <https://arxiv.org/pdf/2005.11650.pdf>`_
+
+    Args:
+        nnodes (int): Number of nodes in the graph.
+        k (int): Number of largest values to consider in constructing the neighbourhood of a node (pick the "nearest" k nodes).
+        dim (int): Dimension of the node embedding.
+        alpha (float, optional): Tanh alpha for generating adjacency matrix, alpha controls the saturation rate, default 3.
+        xd (int, optional): Static feature dimension, default None.
+    """
+
+    def __init__(self, nnodes: int, k: int, dim: int, alpha: float=3, xd: Optional[int]=None):
+        super(GraphConstructor, self).__init__()
+        if xd is not None:
+            self._static_feature_dim = xd
+            self._linear1 = nn.Linear(xd, dim)
+            self._linear2 = nn.Linear(xd, dim)
+        else:
+            self._embedding1 = nn.Embedding(nnodes, dim)
+            self._embedding2 = nn.Embedding(nnodes, dim)
+            self._linear1 = nn.Linear(dim, dim)
+            self._linear2 = nn.Linear(dim, dim)
+
+        self._k = k
+        self._alpha = alpha
+
+    def forward(self, idx: torch.LongTensor, static_feat: Optional[torch.FloatTensor]=None) -> torch.FloatTensor:
+        """
+        Making a forward pass to construct an adjacency matrix from node embeddings.
+
+        Arg types:
+            * **idx** (Pytorch Long Tensor) - Input indices, a permutation of the number of nodes, default None (no permutation).
+            * **static_feat** (Pytorch Float Tensor, optional) - Static feature, default None.
+        Return types:
+            * **adj** (PyTorch Float Tensor) - Adjacency matrix constructed from node embeddings.
+        """
+
```

Comment:
Rename matrices to be capital and use something like _layer_name_1 and layer_name_2.